atmo/jobs/forms.py

```diff
@@ -153,12 +154,38 @@ class EditSparkJobForm(BaseSparkJobForm):
                   'extension .ipynb.'
     )
 
+    start_date = forms.DateTimeField(
+        required=True,
+        widget=forms.DateTimeInput(attrs={
+            'class': 'datetimepicker',
+        }),
+        label='Job start date',
+        help_text=('Date and time on which to enable the scheduled Spark job. '
+                   'Changing this field will reset the job schedule.'),
+    )
+
     def __init__(self, *args, **kwargs):
         super(EditSparkJobForm, self).__init__(*args, **kwargs)
         self.fields['notebook'].help_text += (
             '<br />Current notebook: <strong>%s</strong>' % self.instance.notebook_name
         )
 
+    def save(self):
+        now = timezone.now()
+        if 'start_date' in self.changed_data:
+            if self.cleaned_data['start_date'] > now:
+                # If the start_date changed and it's set in the future
+                # reset the last_run_date so that it runs at start_date.
+                self.instance.last_run_date = None
+                print 'setting to none'
+            else:
+                # If the start_date changed and it's set in the past
+                # set last_run_date to start_date, so that it runs at
+                # interval_in_hours hours from start_date.
+                self.instance.last_run_date = self.cleaned_data['start_date']
```

Comment:
This doesn't seem right to me, and I'm not sure if we should allow past start dates at all, and if we allow them we can't set `last_run_date` with it since we rely on that field to know if a job is expired and if it should be run.

The goal of this is to reset the job so as soon as its start date is changed and we literally reset the clock. That means simply setting `last_run_date` to `None` would work for this. That allows us to have accurate information of when a set of job parameters have run last and the user's expectation is met ("If I set a past start date, the job will be run immediately. But If I set a future date, the job will start at that time.")